ansible/roles/sunbird-system-init/templates/system_init_commands.j2

```diff
@@ -0,0 +1,344 @@
+        getKeycloakToken
+        add_user_response=$(curl -X POST {{proto}}://{{proxy_server_name}}/api/org/v1/member/add \
+                -H 'Cache-Control: no-cache' \
+                -H 'Content-Type: application/json' \
+                -H 'accept: application/json' \
+                -H 'authorization: Bearer {{sunbird_api_auth_token}}' \
+                -H 'x-authenticated-user-token: '"$keycloak_access_token" \
+                -d '{
+                        "request": {
+                                "organisationId": "'"$sunbird_custodian_tenant_id"'",
+                                "userId": "'"$sunbird_root_user_id"'",
+                                "roles": ["ORG_ADMIN"]
+                        }
+                    }')
+}
+
+assignRoleToRootUser() {
+        getKeycloakToken
+        assign_role_response=$(curl -X POST {{proto}}://{{proxy_server_name}}/api/user/v1/role/assign \
+                -H 'Cache-Control: no-cache' \
+                -H 'Content-Type: application/json' \
+                -H 'accept: application/json' \
+                -H 'authorization: Bearer {{sunbird_api_auth_token}}' \
+                -H 'x-authenticated-user-token: '"$keycloak_access_token" \
+                -d '{
+                        "request": {
+                                "organisationId": "'"$sunbird_custodian_tenant_id"'",
+                                "userId": "'"$sunbird_root_user_id"'",
+                                "roles": ["ORG_ADMIN"]
+                        }
+                    }')
+}
+
+updateSettingsCustodianOrgCreated() {
+        system_settings_id="custodianOrgCreated";
+        system_settings_value="true";
+        setValueInSystemSettings;
+}
+
+updateSettingsCustodianOrgId() {
+        system_settings_id="custodianOrgId";
+        system_settings_value="$sunbird_custodian_tenant_id";
+        setValueInSystemSettings;
+}
+
+updateSettingsCustodianOrgChannel() {
+        system_settings_id="custodianOrgChannel";
+        system_settings_value="{{sunbird_custodian_tenant_channel}}";
+        setValueInSystemSettings;
+}
+
+updateSettingsSystemInit() {
+        system_settings_id="systemInit";
+        system_settings_value="true";
+        setValueInSystemSettings;
+}
+
+updateSettingsChannelRegistered() {
+        system_settings_id="custodianOrgChannelRegistered";
+        system_settings_value="true";
+        setValueInSystemSettings;
+}
+
+updateSettingsHashTagRegistered() {
+        system_settings_id="custodianOrgHashTagRegistered";
+        system_settings_value="true";
+        setValueInSystemSettings;
+}
+
+updateSettingsRootUserCreated() {
+        system_settings_id="custodianOrgRootUserCreated";
+        system_settings_value="true";
+        setValueInSystemSettings;
+}
+
+system_settings_id="systemInit";
+getValueFromSystemSettings;
+if [ "$system_settings_value" = "false" ];
+then
+        echo "Starting System Initialisation...";
+        validateConfigFields;
```

Comment:
validateConfigFields didn't return an error even though I didn't define any of the following variables.

sunbird_custodian_tenant_name
sunbird_custodian_tenant_description
sunbird_custodian_tenant_channel

sunbird_root_user_firstname
sunbird_root_user_username
sunbird_root_user_password
sunbird_root_user_email
sunbird_root_user_phone

Reply:
Review Comment Incorporated in code